RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  PurchaseButton.swift
+//  
+//
+//  Created by Nacho Soto on 7/18/23.
+//
+
+import RevenueCat
+import SwiftUI
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+struct PurchaseButton: View {
+
+    let package: Package
+    let purchaseHandler: PurchaseHandler
+    let colors: PaywallData.Configuration.Colors
+    let localization: ProcessedLocalizedConfiguration
+    let introEligibility: IntroEligibilityStatus?
+    let mode: PaywallViewMode
+
+    @State
+    private var purchased = false
+    @Environment(\.dismiss)
+    private var dismiss
+
+    var body: some View {
+        // Workaround for SwiftUI bug
```

Comment:
TODO: file a Radar for https://twitter.com/nachosoto/status/1681447887077801989?s=61&t=xbQZk0UV1WUriSSGU7EvJw